docs/template_plugin/tests/functional/op_reference/base_reference_test.cpp

```diff
@@ -0,0 +1,173 @@
+    ASSERT_EQ(inputInfo.size(), inputData.size());
+
+    for (size_t i = 0; i < params.size(); i++) {
+        const auto& param = params[i];
+        const auto infoIt = inputInfo.find(param->get_friendly_name());
+        GTEST_ASSERT_NE(infoIt, inputInfo.cend());
+
+        const auto& info = infoIt->second;
+        auto blob = make_blob_with_precision(info->getTensorDesc());
+        blob->allocate();
+
+        ASSERT_EQ(blob->byteSize(), inputData[i]->byteSize());
+
+        MemoryBlob::Ptr mInputData = as<MemoryBlob>(inputData[i]);
+        ASSERT_NE(mInputData, nullptr);
+        auto minputDataHolder = mInputData->rmap();
+
+        MemoryBlob::Ptr mBlob = as<MemoryBlob>(blob);
+        ASSERT_NE(mBlob, nullptr);
+        auto mBlobHolder = mBlob->wmap();
+
+        std::memcpy(mBlobHolder.as<void*>(), minputDataHolder.as<const void*>(), inputData[i]->byteSize());
+        inputData[i] = blob;
+    }
+}
+
+void CommonReferenceTest::Infer() {
+    inferRequest = executableNetwork.CreateInferRequest();
+
+    const auto& inputsInfo = executableNetwork.GetInputsInfo();
+    const auto& functionParams = function->get_parameters();
+    for (int i = 0; i < functionParams.size(); ++i) {
+        const auto& param = functionParams[i];
+        const auto infoIt = inputsInfo.find(param->get_friendly_name());
+        GTEST_ASSERT_NE(infoIt, inputsInfo.cend());
+
+        const auto& info = infoIt->second;
+        auto blob = inputData[i];
+
+        inferRequest.SetBlob(info->name(), blob);
+    }
+    inferRequest.Infer();
+}
+
+void CommonReferenceTest::Compare() {
+    ASSERT_EQ(executableNetwork.GetOutputsInfo().size(), refOutData.size());
+    std::vector<InferenceEngine::Blob::Ptr> outputs;
+    for (const auto& result : function->get_results()) {
+        auto name = ngraph::op::util::create_ie_output_name(result->input_value(0));
+        outputs.emplace_back(inferRequest.GetBlob(name));
+    }
+
+    ASSERT_EQ(refOutData.size(), outputs.size());
+    for (size_t i = 0; i < refOutData.size(); i++) {
+        CompareBlobs(refOutData[i], outputs[i]);
+    }
+}
+void CommonReferenceTest::CompareBlobs(const InferenceEngine::Blob::Ptr& refBlob, const InferenceEngine::Blob::Ptr& outBlob) {
+    ASSERT_TRUE(refBlob != nullptr);
+    ASSERT_TRUE(outBlob != nullptr);
+    ASSERT_EQ(refBlob->getTensorDesc().getPrecision(), outBlob->getTensorDesc().getPrecision());
+    ASSERT_EQ(refBlob->byteSize(), outBlob->byteSize());
+
+    auto mRef = as<InferenceEngine::MemoryBlob>(refBlob);
+    IE_ASSERT(mRef);
+    const auto refLockMemory = mRef->rmap();
+    const auto refBuffer = refLockMemory.as<const std::uint8_t*>();
+
+    auto mOut = as<InferenceEngine::MemoryBlob>(outBlob);
+    IE_ASSERT(mOut);
+    const auto outLockMemory = mOut->rmap();
+    const auto outBuffer = outLockMemory.as<const std::uint8_t*>();
+
+    const auto& precision = refBlob->getTensorDesc().getPrecision();
+    switch (precision) {
```

Comment:
1) IMHO this `switch` should be put to separate function. This is technical detail how buffers are "compare".
2) `Compare` doesn't compare - no output value plus side effects - it check if blobls are equal.

Reply:
Renamed method to `Validate`